src/mame/drivers/gsz80.cpp

```diff
@@ -38,13 +41,54 @@ class gsz80_state : public driver_device
 	required_device<acia6850_device> m_acia;
 };
 
+class rc2014mini_state : public gsz80_state
+{
+public:
+	rc2014mini_state(const machine_config &mconfig, device_type type, const char *tag)
+		: gsz80_state(mconfig, type, tag)
+		, m_rombank(*this, "bank")
+		, m_jump_rom(*this, "A13-15")
+	{ }
+
+	// Different machine config due to different RAM
+	void rc2014mini(machine_config &config);
+
+protected:
+	// RC2014 Mini only has 32K RAM, so memory map is different
+	void rc2014mini_mem(address_map &map);
+
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+
+	required_memory_bank m_rombank;
+	required_ioport m_jump_rom;
+};
+
 // Trivial memory map for program memory
 void gsz80_state::gsz80_mem(address_map &map)
 {
 	map(0x0000, 0x1fff).rom();
 	map(0x2000, 0xffff).ram();
 }
 
+void rc2014mini_state::machine_start()
+{
+	m_rombank->configure_entries(0, 8, memregion("maincpu")->base(), 0x2000);
+}
+
+// Set ROM bank from machine CONF at Reset
+void rc2014mini_state::machine_reset()
+{
+	m_rombank->set_entry(m_jump_rom->read() & 7);
+}
+
+// RC2014 Mini only has 32K RAM
+void rc2014mini_state::rc2014mini_mem(address_map &map)
+{
+	map(0x0000, 0x1fff).rom().bankr("bank");
```

Comment:
You shouldn’t have the `.rom()` in there – the `.bankr("bank")` is overriding it.  It’s confusing like this.  It should just have the `.bankr("bank")`